docs/reference/ingress/kubernetes-with-cert-manager.md

```diff
@@ -0,0 +1,168 @@
+
+**RBAC** We will assume that your cluster is configured for [RBAC][k8s-rbac].
+
+## Configure Helm and Tiller
+
+First install Helm and the Tiller [following the instructions provided by Helm][helm-install]
+
+## Create namespaces
+
+```sh
+$ kubectl create namespace openfaas
+$ kubectl create namespace openfaas-fn
+```
+
+## Install nginx-ingress
+
+Add the `nginx-ingress` using
+
+```sh
+$ helm install stable/nginx-ingress --name nginxingress --set rbac.create=true
+```
+
+The full configuration options for nginx [can be found here][nginx-configuration].
+
+## Install cert-manager
+
+We will treat `cert-manager` like a core component of the cluster, so we install it into the `kube-system` namespace using the following command:
+
+```sh
+$ helm install \
+    --name cert-manager \
+    --namespace kube-system \
+    stable/cert-manager
+```
+
+This configuration will work for most deployments, but you can also see https://cert-manager.readthedocs.io/en/latest/getting-started/2-installing.html for additional instructions and options for installing cert-manager.
+
+## Configure cert-manager
+
+In additional to the controller installed in the previous step, we must also configure an "Issuer" before `cert-manager` can create certificates for our services. For convenience we will create an Issuer for both Let's Encrypt's production API and their staging API. The staging API has much kinder rate limits and is useful for testing.
+
+Replace `<your-email-here>` with the contact email that will be shown with the SSL certificate.
+
+```yaml
+# letsencrypt-issuer.yaml
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-prod
+  namespace: openfaas
+spec:
+  acme:
+    # Email address used for ACME registration
+    email: <your-email-here>
+    http01: {}
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      key: ""
+      name: letsencrypt-prod
+    server: https://acme-v01.api.letsencrypt.org/directory
+---
+apiVersion: certmanager.k8s.io/v1alpha1
+kind: Issuer
+metadata:
+  name: letsencrypt-staging
+  namespace: openfaas
+spec:
+  acme:
+    server: https://acme-staging.api.letsencrypt.org/directory
+    # Email address used for ACME registration
+    email: <your-email-here>
+    # Name of a secret used to store the ACME account private key
+    privateKeySecretRef:
+      name: letsencrypt-staging
+    http01: {}
+```
+
+```sh
+$ kubectl apply -f letsencrypt-issuer.yaml
+```
+
+This will allow `cert-manager` to automatically provision Certificates just in the `openfaas` namespace.
+
+## Install openfaas
+
+The OpenFaaS Helm Chart already supports the nginx-ingress, but we want to customize it further. This is easiest with a custom values file. Below, we enable and configure the ingress object to use our certificate and expose just the gateway
+
+```yaml
+# our-values.yaml
```

Comment:
Is `our-values.yaml` a good name? What about something like what it is?

`tls.yml` or similar ?

Reply:
`tls.yaml` sounds good.  I specifically went with a bad name because I wasn't sure which one to use.